components/script/dom/html/htmlmediaelement.rs

```diff
@@ -2252,26 +2249,22 @@ impl HTMLMediaElement {
     }
 
     fn playback_need_data(&self, can_gc: CanGc) {
-        // The player needs more data.
-        // If we already have a valid fetch request, we do nothing.
-        // Otherwise, if we have no request and the previous request was
-        // cancelled because we got an EnoughData event, we restart
-        // fetching where we left.
-        if let Some(ref current_fetch_context) = *self.current_fetch_context.borrow() {
-            if let Some(reason) = current_fetch_context.cancel_reason() {
+        // The player needs more data. If we already have a valid fetch request, we do nothing.
+        // Otherwise, if we have no request and the previous request was cancelled because we got an
+        // EnoughData event, we restart fetching where we left.
+        if let Some(ref mut current_fetch_context) = *self.current_fetch_context.borrow_mut() {
+            if let Some(cancel_reason) = current_fetch_context.cancel_reason() {
```

Comment:
As far as I can see, the changes to this part of the code are not required to solve the original issue, right? If that's true, I would prefer to revert them as well as the changes to make `current_fetch_context` use `#[no_trace]`, because I'm still not confident that that's a good long-term solution.